CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKIconTextureAtlas.java

```diff
@@ -219,16 +226,46 @@ public void IncrementDescriptorTypeRequirements(CVKDescriptorPoolRequirements re
     
     @Override
     public int DisplayUpdate() {
-        Destroy();
-        return CreateAtlas();
+        int ret = VK_SUCCESS;
+        
+        if (needsSaveToFile) {
+            ret = SaveToFile(fileToSave);
+            needsSaveToFile = false;
+        }
+        
+        if (needsAtlasRebuilt) {
+            Destroy();
+            ret = CreateAtlas();
+            needsAtlasRebuilt = false;
```

Comment:
This needs a return if ret failed before resetting needsAtlasRebuilt.

Reply:
changed